core/src/test/java/org/elasticsearch/search/suggest/SuggestBuilderTests.java

```diff
@@ -60,6 +131,31 @@ protected NamedWriteableRegistry provideNamedWritableRegistry() {
     }
 
     /**
+     * Test that a malformed JSON completion suggestion request fails.
+     */
+    public void testMalformedJsonRequestPayload() throws Exception {
+        final String field = RandomStrings.randomAsciiOfLength(getRandom(), 10).toLowerCase(Locale.ROOT);
+        final XContentBuilder suggest = jsonBuilder().startObject()
+                                            .startObject("bad-payload")
+                                            .field("prefix", "sug")
+                                            .startObject("completion")
+                                            .field("field", field)
+                                            .startArray("payload")
+                                            .startObject()
+                                            .field("payload", "field")
+                                            .endObject()
+                                            .endArray()
+                                            .endObject()
+                                            .endObject().endObject();
+        try {
+            final SuggestBuilder suggestBuilder = SuggestBuilder.fromXContent(newParseContext(suggest), suggesters);
+            fail("Should not have been able to create SuggestBuilder from malformed JSON: " + suggestBuilder);
+        } catch (IllegalArgumentException e) {
+            assertThat(e.getMessage(), equalTo("Suggestion must have name"));
```

Comment:
I just took a look at the original test, I think it is testing syntax error in `"payload":[{"payload":"field"}]` (according to the docs it should be `"payload": ["payload":"field"]", the error you are getting now is hiding that. In order to get rid of the required name error I think you can add another object around the existing one. Maybe it would be easier to read the json if you could change the test so the malformed json is specified as a String (see e.g. HighlightBuilderTests#testUnknownArrayNameExpection for examples)